src/wire/libevent_callbacks.cpp

```diff
@@ -152,10 +152,14 @@ void StateMachine(LibeventSocket *conn) {
           conn->pkt_manager.is_started = true;
         } else {
           // Process all other packets
-          status = conn->pkt_manager.ProcessPacket(&conn->rpkt, (size_t)conn->thread_id);
+          status = conn->pkt_manager.ProcessPacket(conn);
         }
-
-        if (status == false) {
+        if (conn->worker_executing){
```

Comment:
Why would it fall into this state while the 'worker_executing' thread is still false? Shouldn't it be set to true in "void PacketManager::ExecExecuteMessageAsync(void* conn_ptr)" already?